new_study_assistant.py

```diff
@@ -0,0 +1,624 @@
+                    table.add_hline()
+                    table.add_hline()
+
+                for row in predicted_types.index:
+                    table.add_row(list(predicted_types.loc[row,:]))
+                    table.add_hline()
+
+        doc.append(NoEscape(r'\newpage'))
+
+        with doc.create(Subsection('Number of attribute distributions')):
+            with doc.create(Figure(position='h!')) as n_attributes:
+                n_attributes.add_image(file_directory + 'n_attribute_distribution.png', width=NoEscape(r'0.99\textwidth'))
+                n_attributes.add_caption('Comparison between the number of attributes in the new study and the number of attributes in each existing study on cBioPortal.  The dashed black line indicates the number of attributes in the new study, while the histogram shows the data for existing cBioPortal studies.')
+
+        #with doc.create(Subsection('Number of unique attributes')):
+            with doc.create(Figure(position='h!')) as n_unique_attributes:
+                n_unique_attributes.add_image(file_directory + 'n_unique_attribute_distribution.png', width=NoEscape(r'0.99\textwidth'))
+                n_unique_attributes.add_caption('Comparison between the number of unique attributes in the new study and the number of unique attributes in each existing study on cBioPortal.  The dashed black line indicates the number of unqiue attributes in the new study, while the histogram shows the data for existing cBioPortal studies.')
+
+        #with doc.create(Subsection('Number of common attributes')):
+            with doc.create(Figure(position='h!')) as n_common_attributes:
+                n_common_attributes.add_image(file_directory + 'n_common_attribute_distribution.png', width=NoEscape(r'0.99\textwidth'))
+                n_common_attributes.add_caption('Comparison between the number of common attributes in the new study and the number of common attributes in each existing study on cBioPortal.  The dashed black line indicates the number of common attributes in the new study, while the histogram shows the data for existing cBioPortal studies.')
+
+    doc.generate_pdf('report', clean_tex=True)
+
+def combine_matches(exact, name, value, no_matches):   
+    all_matches = (pd.concat([exact, value, name, no_matches])
+                      .drop_duplicates()
+                      .sort_values('New study attribute',axis=0)
+                      .reset_index(drop=True))
+    
+    for index, row in all_matches.iterrows():
+        append_string = ''
+        if (sum(exact['Possible matches'] == row['Possible matches']) > 0) or (sum(name['Possible matches'] == row['Possible matches']) > 0):
+            append_string += '^'
+        if sum(value['Possible matches'] == row['Possible matches']) > 0:
+            append_string += '*'
+        else:
+            append_string += ''
+
+        if row['Possible matches'] != 'No matches found':            
```

Comment:
I think I see now that no_matches are records indicating that a new study attribute has no matches ("No matches found") .. so calling the combination "all_matches" is a little confusing since some of the "matches" are anti-matches. Maybe a name like "all_match_results" would be more clear.